puepy/core.py

```diff
@@ -311,20 +315,30 @@ def _render_onto(self, element, attrs):
 
         # Add bind
         if self.bind and self.origin:
-            if _element_input_type(element) == "checkbox":
+            input_type = _element_input_type(element)
+            if input_type == "checkbox":
                 if is_server_side and self.origin.state[self.bind]:
                     element.setAttribute("checked", self.origin.state[self.bind])
                 else:
                     element.checked = bool(self.origin.state[self.bind])
                     element.setAttribute("checked", self.origin.state[self.bind])
+                event_type = "change"
+            elif input_type == "radio":
+                is_checked = (self.origin.state[self.bind] == element.value)
+                if is_server_side and is_checked:
+                    element.setAttribute("checked", is_checked)
+                else:
+                    element.checked = is_checked
+                    element.setAttribute("checked", is_checked)
+                event_type = "change"
             else:
                 if is_server_side:
                     element.setAttribute("value", self.origin.state[self.bind])
                 else:
                     element.value = self.origin.state[self.bind]
                     element.setAttribute("value", self.origin.state[self.bind])
-
-            self._add_event_listener(element, "input", self.on_bind_input)
+                event_type = "change"
```

Comment:
This actually changes the input type of other inputs from `input` to `change`, which has some unexpected consequences. I'm inclined to approve the PR, and I like what you've done here, but I'll probably change that back and tinker with it. (It does break the playwright tests)